components/director/internal/domain/application/service.go

```diff
@@ -764,6 +767,13 @@ func (s *service) Merge(ctx context.Context, destID, srcID string) (*model.Appli
 		return nil, errors.Errorf("Application templates are not the same. Destination app template: %s. Source app template: %s", destTemplateID, srcTemplateID)
 	}
 
+	if _, exists := srcAppLabels[s.selfRegisterDistinguishLabelKey]; exists {
```

Comment:
I think that you have to check apptemplate labels instead of srcAppLabes?

Reply:
Indeed, the app template will be the one that can have the distinguished label, so you should check that instead of the srcApp/destApp labels.